kernel/USER_GUIDE.md

```diff
@@ -494,33 +507,64 @@ import io.delta.kernel.utils.*;
 Byte[] scanStateRowBytes = RowUtils.serialize(scanStateRow);
 Byte[] scanFileRowBytes = RowUtils.serialize(scanFileRow);
 
+// Optionally the connector add a split info to the task (scan file, scan state) to
+// split reading of the Parquet files into multiple tasks. The task gets split info
+// along with the scan file row and scan state row.
+Split split = ...; // connector specific class, not related to Kernel
+
 // Send these over to the worker
 
 // In the worker when data will be read, after rowBytes have been sent over
-Row scanStateRow = RowUtils.deserialize(scanStateRowBytes)
-Row scanFileRow = RowUtils.deserialize(scanFileRowBytes)
+Row scanStateRow = RowUtils.deserialize(scanStateRowBytes);
+Row scanFileRow = RowUtils.deserialize(scanFileRowBytes);
+Split split = ... deserialize split info ...;
 ```
 
 #### Step 3.4: Read the columnar data 
 Finally, we are ready to read the columnar data. You will have to do the following:
 
-* Get the iterator of `ColumnarBatch`es: Use the scan state row and scan file row(s) to call the following to get a new iterator of data batches.
+* Read the physical data from Parquet file as indicated by the scan file row, scan state, and optionally the split info
+* Convert the physical data into logical data of the table using the Kernel's APIs.
 
 ```java
-Row myScanStateRow = ... ;
-Iterator<Row> myScanFileRows = ... ;
+Row scanStateRow = ... ;
+Row scanFileRow = ... ;
+Split split = ...;
 
 // Additional option predicate such as dynamic filters the connector wants to
 // pass to the reader when reading files.
 Predicate optPredicate = ...;
 
-CloseableIterator<FilteredColumnarBatch> dataIter = Scan.getData(
-  myTableClient,
-  myScanStateRow,
-  myScanFileRows,
-  Optional<>.empty(optPredicate)
+// Get the physical read schema of columns to read from the Parquet data files
+StructType physicalReadSchema =
+  ScanStateRow.getPhysicalDataReadSchema(tableClient, scanStateRow);
+
+// From the scan file row, extract the file path, size and modification metadata
+// needed to read the file.
+FileStatus fileStatus = InternalScanFileUtils.getAddFileStatus(scanFileRow);
+
+// Open the scan file which is a Parquet file using connector's own
+// Parquet reader which supports reading specific parts (split) of the file.
+// If the connector doesn't have its own Parquet reader, it can use the
+// default Parquet reader provider which at the moment doesn't support reading
+// a specific part of the file, but reads the entire file from the beginning.
+CloseableIterator<ColumnarBatch> physicalDataIter =
+  connectParquetReader.readParquetFile(
+  fileStatus
+  physicalReadSchema,
+  split, // what part of the Parquet file to read data from
+  optPredicate /* additional predicate the connector can apply to filter data from the reader */
 );
 
+// Now the physical data read from the Parquet data file is converted to a table
```

Comment:
"a table logical data"?

Reply:
updated to: "to logical data the table represents"